web/models/terms/term.ex

```diff
@@ -31,5 +32,17 @@ defmodule CoursePlanner.Terms.Term do
     |> validate_required([:name, :start_date, :end_date, :status])
     |> cast_embed(:holidays)
     |> Statuses.update_status_timestamp(EntityStatus)
+    |> validate_date_range()
   end
+
+  defp validate_date_range(%{errors: []} = changeset) do
+    st = changeset |> Changeset.get_field(:start_date) |> Date.cast!
```

Comment:
I thought so as well, but I was getting some weird things from the difference between native `Date` (`~D` sigil at least) and `Ecto.Date`, since `Term.changeset/2` is used with both types and this was the solution I found to "normalize" the types.